### DISCUSSION TITLE
Performance issues on new faces

### DISCUSSION THREAD

**Discussion by amgschnappi** [photoprism/photoprism#2285] [General]
I have at least 90K+ photos. After months of indexing, it shows more than 2k new faces. Unfortunately, no browser is capable of loading this page on my modest 10 years old laptop.
Is there any way to increase performance?

**Comment by lastzero** [comment]
Of course the performance CAN be improved. The question is what it costs compared to better hardware. Every vendor has hardware requirements and so do we... 

Besides, 2000 face clusters seems to be way too much unless you only have portraits/faces and from each person many shots? After MONTHS of indexing also sounds much too long, should be days max. Does that mean, you never updated? Our latest build should already be faster.

**Comment by lastzero** [comment]
I'm moving this to GitHub Discussions as it needs a root cause and feasibility analysis before it can be addressed. Our team is constantly working to improve performance based on our own and our users' experience on (probably) more modern hardware. I've also added a general note to our [FAQs in Getting Started](https://docs.photoprism.app/getting-started/faq/#can-you-improve-performance-when-using-older-or-otherwise-slow-hardware).

**Comment by amgschnappi** [comment]
I am using win10 with WSL set to use 2GB max (since my HW has 4 GB). I will move to some linux distro and try this again. Will post the outcome if faces loads successfully without the wsl limit.

**Reply by lastzero** [reply]
If the hardware is already not the fastest, then running it in a virtual environment like Docker on Windows can further reduce performance. Especially if the UI (Web browser) runs on the same hardware, so resources have to be shared between front- and backend.
